src/deriv_api/DerivAPIBasic.js

```diff
@@ -253,16 +268,35 @@ export default class DerivAPIBasic extends DerivAPICalls {
         this.events.next({
             name: 'close',
         });
-
         if (this.shouldReconnect) {
             this.events.next({
                 name: 'reconnecting',
             });
 
-            this.connect();
+            this.reconnect();
         }
     }
 
+    /**
+     * Clears previous connection keeplive ping timeout and connect & assign the handles
+     */
+    reconnect() {
+        this.pong(); // clear all previous timeout
+        this.connect();
```

Comment:
If it reconnect failed, then it should reconnect again.

If the connection is given from argument of `new DerivAPIBasic(...), it shouldn't do reconnecting. The caller of it should handle reconnection itself.